src/_pytest/config/argparsing.py

```diff
@@ -337,14 +336,15 @@ def __init__(self, parser, extra_info=None, prog=None):
         self.extra_info = extra_info
 
     def error(self, message):
-        """error(message: string)
-
-        Prints a usage message incorporating the message to stderr and
-        exits.
-        Overrides the method in parent class to change exit code"""
-        self.print_usage(_sys.stderr)
+        """Transform argparse error message into UsageError."""
         args = {"prog": self.prog, "message": message}
-        self.exit(EXIT_USAGEERROR, _("%(prog)s: error: %(message)s\n") % args)
+        msg = _("%(prog)s: error: %(message)s\n") % args
+        msg = msg.rstrip("\n")  # to not require re-translation?!
```

Comment:
We're not using gettext translations, are we?
Then the "\n" in the msg could be dropped already.

Reply:
Found the reason: https://github.com/pytest-dev/pytest/pull/3925/files#r215195026.
I think it can be removed then.

Reply:
Indeed.